File: oop/02_base/03_constructor.py
class Student:
    def __init__(self, name, last_name):
        self.name = name
        self.last_name = last_name

    def get_full_name(self):
        print("Полное имя: " + self.name + ' ' + self.last_name)


student1 = Student('Сергей', 'Иванов')
student1.get_full_name()

student2 = Student('Мария', 'Иванова')
student2.get_full_name()

File: OOP/02_base/03_constructor.php
<?php
class Student
{
    public string $name;
    public string $lastName;

    // signatura: __construct($this, $name, $lastName)
    public function __construct(string $name, string $lastName)
    {
        $this->name = $name;
        $this->lastName = $lastName;
    }

    public function showFullName()
    {
        echo "Полное имя: ".$this->name." ".$this->lastName.PHP_EOL;
    }
}

$student1 = new Student("Сергей", "Иванов");
$student1->showFullName();

$student2 = new Student("Мария", "Петрова");
$student2->showFullName();
